libimage/layer_tree.go

```diff
@@ -91,19 +91,38 @@ func (l *layerNode) repoTags() ([]string, error) {
 
 // layerTree extracts a layerTree from the layers in the local storage and
 // relates them to the specified images.
-func (r *Runtime) layerTree(ctx context.Context, images []*Image) (*layerTree, error) {
-	layers, err := r.store.Layers()
+// layerTree require both parameters not empty or nil, to avoid inconsistences
+// between layers and images. If both parameters are nil then layers and images will be obtained from local storage.
+// You can obtain consistent layer and images with storage.MultiList.
+func (r *Runtime) layerTree(images []*Image, layers []storage.Layer) (*layerTree, error) {
+	if (images == nil && layers != nil) || (images != nil && layers == nil) {
+		return nil, errors.New("layerTree require both parameters not empty or nil, to avoid inconsistences between layers and images.")
+	}
+
+	optionsList := storage.MultiListOptions{}
+	if images == nil {
+		optionsList.Images = true
+	}
+
+	if layers == nil {
+		optionsList.Layers = true
+	}
```

Comment:
The code just above has enforced that `images == nil` == `layers == nil`; so this code does not need to handle inconsistent combinations; in fact we _explicitly don’t want_ inconsistent combinations.

More to the point, if the data is available already, we don’t need to call `MultiList(MultiListOptions{})`.

… and if we only call `MultiList` because we don’t have the data, the code just below to obtain the data from `listResult` won’t need to be separately conditional either.